petab/v2/problem.py

```diff
@@ -981,3 +976,49 @@ def add_experiment(self, id_: str, *args):
             if self.experiment_df is not None
             else tmp_df
         )
+
+
+class ModelFile(BaseModel):
+    """A file in the PEtab problem configuration."""
+
+    location: str | AnyUrl
+    language: str
+
+
+class SubProblem(BaseModel):
+    """A `problems` object in the PEtab problem configuration."""
+
+    model_files: dict[str, ModelFile] | None = {}
+    measurement_files: ListOfFiles = []
+    condition_files: ListOfFiles = []
+    experiment_files: ListOfFiles = []
+    observable_files: ListOfFiles = []
+    visualization_files: ListOfFiles = []
+    mapping_files: ListOfFiles = []
+
+
+class ExtensionConfig(BaseModel):
+    """The configuration of a PEtab extension."""
+
+    name: str
+    version: str
+    config: dict
+
+
+class ProblemConfig(BaseModel):
+    """The PEtab problem configuration."""
+
+    filepath: str | AnyUrl | None = Field(
+        None,
+        description="The path to the PEtab problem configuration.",
+        exclude=True,
+    )
+    base_path: str | AnyUrl | None = Field(
```

Comment:
I prefer just `root` but fine as is

Reply:
My association with `root` would be that the other files are somewhere below in the hierarchy, and `root/..` would be illegal. This is not the case here. It might be just my association, though. I am open to change it later on if others will strong about it.

Reply:
`root` is used as the location to which all other things are relative here:
https://docs.python.org/3/library/shutil.html#shutil.make_archive

But I think others will have varying opinions about the difference between `root` and `base` so fine :+1: